moveit_ros/moveit_servo/test/unit_test_servo_calcs.cpp

```diff
@@ -0,0 +1,536 @@
+ *     with the distribution.
+ *   * Neither the name of PickNik LLC nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/*      Title     : unit_test_servo_calcs.cpp
+ *      Project   : moveit_servo
+ *      Created   : 07/24/2020
+ *      Author    : Adam Pettinger
+ *      Desc      : Loads a Servo instance without launching or parameters
+ */
+
+#include <gtest/gtest.h>
+#include <moveit_servo/servo.h>
+
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <boost/filesystem/path.hpp>
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include "test_parameter_struct.hpp"
+#include "unit_test_servo_calcs.hpp"
+
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_servo.unit_test_servo_calcs.cpp");
+
+void loadModelFile(std::string filename, std::string& file_content)
+{
+  boost::filesystem::path res_path(ament_index_cpp::get_package_share_directory("moveit_resources"));
+  std::string xml_string;
+  std::fstream xml_file((res_path / filename).string().c_str(), std::fstream::in);
+  // EXPECT_TRUE(xml_file.is_open());
+  EXPECT_TRUE(5 == 5);
+  while (xml_file.good())
+  {
+    std::string line;
+    std::getline(xml_file, line);
+    xml_string += (line + "\n");
+  }
+  xml_file.close();
+  file_content = xml_string;
+}
+
+FriendServoCalcs::FriendServoCalcs(const rclcpp::Node::SharedPtr& node,
+                                   const moveit_servo::ServoParametersPtr& parameters,
+                                   const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor)
+  : ServoCalcs(node, parameters, planning_scene_monitor)
+{
+}
+
+ServoCalcsTestFixture::ServoCalcsTestFixture() : node_(std::make_shared<rclcpp::Node>("servo_calcs_test"))
+{
+  // "Load" parameters from yaml as node parameters
+  std::string robot_description_string, srdf_string;
+  loadModelFile("panda_description/urdf/panda.urdf", robot_description_string);
```

Comment:
Better use the convenience functions from `moveit_test_utils`

Reply:
Servo is looking for the URDF and SRDF as parameters (strings). To set that for testing I was doing:

```
loadModelFile("panda_description/urdf/panda.urdf", robot_description_string);
loadModelFile("panda_moveit_config/config/panda.srdf", srdf_string);
node_->declare_parameter<std::string>("robot_description", robot_description_string);
node_->declare_parameter<std::string>("robot_description_semantic", srdf_string);
```
I don't see a way in the `moveit_test_utils` to give the path to the file and get back the string for loading as a parameter, but maybe I am missing something?

Reply:
If you only want to use a `moveit_resources` package `moveit_test_utils` is the way to go. If you want to support any robot using the `robot_description` parameter, the PSM (or the RobotModelLoader) supports this without anything extra and the `moveit_resources` default should be passed with the launch file. If your actual use case is not covered, maybe it should go into `moveit_test_utils`.

Reply:
I see, `moveit_resources` doesn't really help because the PSM expects either the robot_description parameter or a `RobotModelLoader` (not the RobotModel). Maybe it would still be a good idea to move a convenience function for this to test utils. Something that declares the robot description parameters from a default package and optionally provides you with the loader instance.